src/pages/index.js

```diff
@@ -1,168 +1,241 @@
-
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-
-  handleProfileEditSubmit
-);
-
-profileEditPopup.setEventListeners();
+function createCard(cardData) {
+  const card = new Card(
+    cardData.name,
+    cardData.link,
+    cardData.isLiked,
+    cardData.likes,
+    cardData._id,
+    userId,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector
+  );
 
-profileEditButton.addEventListener("click", () => {
-  handleProfileEditClick();
-});
+  return card.getView();
+}
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false, "Save");
+    });
+  console.log(inputValues);
+}
+function handleCardAddClick(inputValues) {
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard(inputValues)
+    .then((cardData) => {
+      const addCard = createCard(cardData);
+      newCardSection.addItem(addCard);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false, "Create");
+    });
+}
+function handleAviPopup(inputValues) {
+  avatarPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatar(inputValues.avatar);
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarPopup.renderLoading(false);
+    });
+}
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(card.cardId)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        deleteCardPopup.renderLoading(false);
+      });
+  });
+}
 
 function handleProfileEditClick() {
   const user = userInfo.getUserInfo();
-
   userNameInput.value = user.name;
-
-  userDescriptionInput.value = user.description;
-
+  userDescriptionInput.value = user.about;
   profileEditPopup.open();
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  profileEditPopup.close();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api
+      .removeCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .addCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors